src/features/projects/projectsSlice.js

```diff
@@ -109,6 +119,37 @@ export const projectsSlice = createSlice({
       state.unsavedViewChanges = payload;
     },
 
+    createProjectStart: (state) => {
+      const ls = { isLoading: true, operation: 'fetching', errors: null, stateMsg: null };
+      state.loadingStates.createProject = ls;
+    },
+
+    createProjectSuccess: (state, { payload }) => {
+      const { project } = payload.createProject;
+      const ls = {
+        isLoading: false,
+        operation: null,
+        errors: null,
+        stateMsg: `Successfully created project ${project.name}`
+      };
+      state.loadingStates.createProject = ls;
+
+      state.projects = {
+        ...state.projects,
+        [project._id]: project
+      }
+    },
+
+    createProjectFailure: (state, { payload }) => {
+      const ls = { isLoading: false, operation: null, errors: payload, stateMsg: null };
```

Comment:
I actually just ran into an error and wasn't alerted. I think we need to wire up error alerts here (sorry in advance for that component being convoluted.. if it's unintelligible I can take this on).

Separately, the error I encountered was thrown because the API requires that at least one model is passed in as an `availableMLModel` and currently the form allows you to submit without any selected models. We should definitely wire up the error alerts for other unexpected errors that may occur, but I'm also confused why the `Yup` validation didn't seem to work for that field when it was empty and it allowed me to submit the form anyhow. Perhaps it's because its a multi select? Any thoughts?

Reply:
I added the error handling.

The ml-models field didn't validate because for Yup, an empty array fulfils the `required` condition, I've added an extra `.min(1)` so you won't be able to submit the form unless at least one model is selected